server/authz.go

```diff
@@ -0,0 +1,44 @@
+package server
+
+import (
+	"context"
+
+	grpc_middleware "github.com/grpc-ecosystem/go-grpc-middleware"
+	grpc "google.golang.org/grpc"
+	credentials "google.golang.org/grpc/credentials"
+	peer "google.golang.org/grpc/peer"
+)
+
+// addUserContext parses client ID from context and set client ID in context
+func addUserContext(ctx context.Context) context.Context {
+	p, ok := peer.FromContext(ctx)
+	if ok {
+		if p.AuthInfo == nil {
+			return ctx
+		}
+		tlsInfo := p.AuthInfo.(credentials.TLSInfo)
+		clientName := tlsInfo.State.VerifiedChains[0][0].Subject.CommonName
```

Comment:
Added a verification for `State.VerifiedChains[0][0]`. In case if that object is also `nil`, which means `*x509.Certificate` pointer is nil, then it is another story. However, if such case happens, it is an edge case that is out of our scope. I.e: it should be a problem of `crypto` package.